poweremail_mailbox.py

```diff
@@ -260,6 +260,18 @@ def complete_mail(self, cr, uid, ids, context=None):
             self.pool.get('poweremail.core_accounts').get_fullmail(cr, uid, id, context)
             self.historise(cr, uid, [id], "Full email downloaded", context)
 
+    def is_valid(self, cursor, uid, mail_id, context=None):
+        fields_to_read = ['pem_to', 'pem_cc', 'pem_bcc']
+        mail = self.read(cursor, uid, mail_id, fields_to_read, context)
+        check_email = self.check_email_valid(mail['pem_to'])
+        if mail['pem_cc']:
+            check_email = check_email and self.check_email_valid(mail['pem_cc'])
+        if mail['pem_bcc']:
+            check_email = check_email and self.check_email_valid(
+                mail['pem_bcc'])
```

Comment:
Why not change all of this code for something like
```python
for field in fields_to_read:
    if mail[field] and not self.check_email_valid(mail[field]):
        return False
return True
```
Using this way the first invalid field return False, the other way we have to check all the fields